src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -48,6 +48,29 @@ class UploadingJobsDisplay extends Component {
     this.uploadJobs();
   }
 
+  componentDidUpdate() {
+    if (!this.props.files) {
```

Comment:
Let's move all this stuff to the method as `componentDidUpdate` is just a hook which can operate on different cases.

Reply:
Updated